docs/reference/commands/keystore.asciidoc

```diff
@@ -0,0 +1,208 @@
+[[elasticsearch-keystore]]
+== elasticsearch-keystore
+
+The `elasticsearch-keystore` command manages <<secure-settings,secure settings>>
+in the {es} keystore.
+
+[discrete]
+[[elasticsearch-keystore-synopsis]]
+=== Synopsis
+
+[source,shell]
+--------------------------------------------------
+bin/elasticsearch-keystore
+([add <setting>] [-f] [--stdin] |
+[add-file <setting> <path>] | [create] [-p] |
+[list] | [passwd] | [remove <setting>] | [upgrade])
+[-h, --help] ([-s, --silent] | [-v, --verbose])
+--------------------------------------------------
+
+[discrete]
+[[elasticsearch-keystore-description]]
+=== Description
+
+IMPORTANT: This command should be run as the user that will run {es}.
+
+Currently, all secure settings are node-specific settings that must have the
+same value on every node. Therefore you must run this command on every node.
+
+When the keystore is password-protected, you must supply the password each time
+{es} starts.
+
+Modifications to the keystore do not take effect until you restart {es}.
+
+Only some settings are designed to be read from the keystore. However, there
+is no validation to block unsupported settings from the keystore and they can
+cause {es} to fail to start. To see whether a setting is supported in the
+keystore, see the setting reference.
+
+[discrete]
+[[elasticsearch-keystore-parameters]]
+=== Parameters
+
+`add <setting>`:: Adds settings to the keystore. By default, you are prompted
+for the value of the setting. If the keystore is password protected, you are
+also prompted to enter the password.
+
+`add-file <setting> <path>`:: Adds a file to the keystore.
+
+`create`:: Creates the keystore.
+
+`-f`:: When used with the `add` parameter, the command overwrites existing
+entries in the keystore.
+//TBD: What happens to existing entries if you do not specify this option? Does the command fail with an error?
```

Comment:
You get a prompt asking if you want to overwrite, e.g.:

```
[vagrant@rhel7 ~]$ sudo /usr/share/elasticsearch/bin/elasticsearch-keystore add bootstrap.password
Setting bootstrap.password already exists. Overwrite? [y/N]
```

Reply:
Thanks for the details!